mpathconfig/configresource.go

```diff
@@ -237,17 +237,33 @@ func addOption(section *Section, option string) {
 	}
 }
 
-func isSection(line string) bool {
-	line = strings.TrimSpace(line)
-	prefixes := []string{"defaults", "blacklist", "blacklist_exceptions", "devices", "device", "multipaths", "multipath"}
-	for _, prefix := range prefixes {
-		if strings.HasPrefix(line, prefix) {
-			return true
-		}
-	}
-	return false
+//checks for the section
+func isSection(line string) (bool, error) {
+        line = strings.TrimSpace(line)
+        prefixes := []string{"defaults", "blacklist", "blacklist_exceptions", "devices", "device", "multipaths", "multipath"}
+        for _, prefix := range prefixes {
+                r, err := regexp.Compile("^"+prefix+"\\s*{")
```

Comment:
Please use this instead of two expressions: ```"^"+prefix+"\s*[{]"``` and add space for ``` err != nil ``` checks. linter didn't catch this?

Reply:
This expression ,"^"+prefix+"\\s*[{]"   won't work in the following case:
multipath
{

Reply:
it does Shubha..https://goplay.space/#ER3D-iC4KoR

Reply:
for each line in multipath.conf  isSection(line) is called .. so isSection("multipath") will be called in one iteration and isSection("\n") in the next iteration...so it won't work....what you said is correct but here since we are calling the isSection with every line so it won't work here.

Reply:
isSection("{") in the next iteration*

Reply:
section name has to be followed by `{` or else multipath will throw an error as below. We can look into automatically handling that separately as i believe good amount of testing is required.

```
devices
{
    device {
        path_selector        "service-time 0"
        fast_io_fail_tmo     5
        prio                 alua
        path_grouping_policy group_by_prio
        vendor               "Nimble"
        path_checker         tur
        rr_min_io_rq         1
        hardware_handler     "1 alua"
        product              "Server"
        no_path_retry        30
        rr_weight            uniform
        failback             immediate
        dev_loss_tmo         infinity
    }
}

# multipath -ll
Oct 15 17:15:29 | missing '{' on line 1 of /etc/multipath.conf
Oct 15 17:15:29 | invalid keyword '{' on line 2 of /etc/multipath.conf```